packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -239,36 +249,46 @@ export const Popover = React.forwardRef((props, forwardedRef) => {
     setPopoverElement,
   );
 
+  const triggerId = `${useId()}-trigger`;
+  const hasAriaLabel = !!props['aria-labelledby'] || !!props['aria-label'];
+
   return (
     <>
       {cloneElementWithRef(children, (children) => ({
+        id: children.props.id || triggerId,
         ...popover.getReferenceProps(children.props),
         ref: popover.refs.setReference,
       }))}
 
       {popover.open ? (
         <Portal portal={portal}>
-          <ThemeProvider
-            portalContainer={popoverElement} // portal nested popovers into this one
-          >
-            <FloatingFocusManager
-              context={popover.context}
-              modal={false}
-              initialFocus={-1}
-              returnFocus
+          <FloatingPortal>
+            <ThemeProvider
+              portalContainer={popoverElement} // portal nested popovers into this one
             >
-              <Box
-                className={cx(
-                  { 'iui-popover-surface': applyBackground },
-                  className,
-                )}
-                {...popover.getFloatingProps(rest)}
-                ref={popoverRef}
+              <FloatingFocusManager
+                context={popover.context}
+                modal={false}
+                initialFocus={popover.refs.floating}
```

Comment:
Would users ever want to override the initial focus? Because in #1406, we are leaning towards asking the users to handle the focus themselves.

Reply:
Resolved. I learned that like the second screen recording above and the `Popover.focus.tsx` example, we can use refs to override the focus.